src/schedule.cpp

```diff
@@ -303,38 +307,100 @@ struct stream_info
     std::unordered_map<instruction_ref, std::unordered_set<instruction_ref>>
     get_conflicts(program& p)
     {
-        std::unordered_map<instruction_ref, std::unordered_set<instruction_ref>> conflict_table;
+        using conflict_table_type =
+            std::unordered_map<instruction_ref, std::unordered_set<instruction_ref>>;
+        conflict_table_type conflict_table;
         auto concur_ins = this->find_concurrent_instructions(p);
+
+        std::unordered_map<std::thread::id, conflict_table_type> thread_conflict_tables;
+        std::mutex mtx;
+        std::vector<instruction_ref> index_to_ins;
         for(auto&& merge : concur_ins)
         {
-            dfor(merge.second.size(), merge.second.size())([&](auto i, auto j) {
-                if(i == j)
-                    return;
-                for(auto ins1 : merge.second[i])
+            index_to_ins.push_back(merge.first);
+        }
+
+        par_for(concur_ins.size(), [&](auto ins_index) {
+            auto merge_first = index_to_ins[ins_index];
+            assert(concur_ins.count(ins1) > 0);
+            auto& merge_second = concur_ins.at(merge_first);
+
+            auto thr_id = std::this_thread::get_id();
+            if(thread_conflict_tables.count(thr_id) == 0)
```

Comment:
The lock would need to include the conditional here.

Reply:
Although, I think it would be better to extend `par_for` to add a sequential ids, and then we can just preallocate a vector.

Reply:
fixed